src/Catalyst.Node.POA.CE.Tests/IntegrationTests/PoaConsensusTests.cs

```diff
@@ -0,0 +1,115 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Autofac;
+using Catalyst.Core.Modules.Consensus.Cycle;
+using Catalyst.Core.Modules.Cryptography.BulletProofs;
+using Catalyst.TestUtils;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Catalyst.Node.POA.CE.Tests.IntegrationTests
+{
+    public sealed class PoaConsensusTests : FileSystemBasedTest
+    {
+        private readonly CancellationTokenSource _endOfTestCancellationSource;
+        private readonly ILifetimeScope _scope;
+        private readonly List<PoaTestNode> _nodes;
+
+        public PoaConsensusTests(ITestOutputHelper output) : base(output)
+        {
+            ContainerProvider.ConfigureContainerBuilder(true, true, true);
+            _scope = ContainerProvider.Container.BeginLifetimeScope(CurrentTestName);
+
+            var context = new FfiWrapper();
+
+            _endOfTestCancellationSource = new CancellationTokenSource();
+
+            var poaNodeDetails = Enumerable.Range(0, 3).Select(i =>
+                {
+                    var privateKey = context.GeneratePrivateKey();
+                    var publicKey = privateKey.GetPublicKey();
+                    var nodeSettings = PeerSettingsHelper.TestPeerSettings(publicKey.Bytes, 2000 + i);
+                    var peerIdentifier = nodeSettings.PeerId;
+                    var name = $"producer{i.ToString()}";
+                    return new {index = i, name, privateKey, nodeSettings, peerIdentifier};
+                }
+            ).ToList();
+
+            var peerIdentifiers = poaNodeDetails.Select(n => n.peerIdentifier).ToList();
+
+            _nodes = poaNodeDetails.Select(
+                p => new PoaTestNode($"producer{p.index.ToString()}",
+                    p.privateKey,
+                    p.nodeSettings,
+                    peerIdentifiers.Except(new[] {p.peerIdentifier}),
+                    FileSystem,
+                    output)).ToList();
+        }
+
+        [Fact]
+        public async Task Run_ConsensusAsync()
+        {
+            _nodes.AsParallel()
+               .ForAll(n =>
+                {
+                    n?.RunAsync(_endOfTestCancellationSource.Token);
+                    n?.Consensus.StartProducing();
+                });
+
+            await Task.Delay(Debugger.IsAttached
+                    ? TimeSpan.FromHours(3)
+                    : CycleConfiguration.Default.CycleDuration.Multiply(1.3))
+               .ConfigureAwait(false);
+
+            var dfsDir = Path.Combine(FileSystem.GetCatalystDataDir().FullName, "dfs");
+            Directory.GetFiles(dfsDir).Length.Should().Be(1,
+                "only the elected producer should score high enough to see his block elected.");
+
+            _endOfTestCancellationSource.CancelAfter(TimeSpan.FromMinutes(3));
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            base.Dispose(disposing);
+            if (!disposing) return;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4594394)